vendor/k8s.io/kube-openapi/pkg/handler3/handler.go

```diff
@@ -192,7 +228,26 @@ func (o *OpenAPIService) HandleGroupVersion(w http.ResponseWriter, r *http.Reque
 			if err != nil {
 				return
 			}
-			w.Header().Set("Etag", etag)
+			// ETag must be enclosed in double quotes: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag
+			w.Header().Set("Etag", strconv.Quote(etag))
+
+			if hash := r.URL.Query().Get("hash"); hash != "" {
+				if hash != etag {
+					u := constructURL(group, etag)
+					http.Redirect(w, r, u, 301)
```

Comment:
we won't serve a request without a hash, but require a redirect and a second request?

Reply:
oh, I misread... this is redirecting a request with a stale hash to the current hash